packages/compartment-mapper/README.md

```diff
@@ -134,6 +134,82 @@ object with an `import({ globals?, modules? })` method.
+to `makeArchive` but generates a string of JavaScript suitable for `eval`.
+We use this "bundle" format to bootstrap an environment up to the point it can
+call `importArchive`, so bundles are at least suitable for creating a script
+that subsumes `ses`, `@endo/compartment-mapper/import-archive.js`, and other
+parts of Endo, but is not as feature-complete as `importArchive`.
+
+```
+import url from 'url';
+import fs from 'fs';
+import { makeBundle } from "@endo/compartment-mapper/bundle.js";
+import { makeReadPowers } from '../node-powers.js';
+const readPowers = makeReadPowers({ fs, url });
+const bundle = await makeBundle(readPowers, moduleLocation, options);
+```
+
+It should be sufficient to evaluate `bundle`.
+The completion value of evaluation is the emulated exports namespace of the
+entry module.
+
+Bundles can include ESM, CJS, and JSON modules, but no other module languages
+like bytes or text.
+
+:warning: Bundles do not support [live
+bindings](https://developer.mozilla.org/en-US/docs/Glossary/Binding), dynamic
+`import`, or `import.meta`.
+Bundles do not isolate modules to a compartment.
+
+`makeBundle` accepts all the options of `makeArchive` and:
+
+- `sourceUrlPrefix` (`string`, default `''`)::
+  Specifies a prefix to occur on each module's `sourceURL` comment, as injected
+  at runtime.
+  Should generally end with `/` if non-empty.
+  This can improve stack traces.
+  Can be overriden with the `sourceUrlPrefix` curried runtime option with
+  `curryRuntimeOptions`.
+- `format` (`"cjs"` or `undefined`, default `undefined`):
+  By default, `makeBundle` generates a bundle that can be evaluated in any
+  context.
+  By specifying `"cjs"`, the bundle can assume there is a host CommonJS
+  `require` function available for resolving modules that exit the bundle.
+  The default is `require` on `globalThis`.
+  The `require` function can be overridden with a curried runtime option.
+- `useEvaluate` (`boolean`, default `false`):
+  By default, a bundle has an array of functions for each bundled module.
+  With `useEvaluate`, the bundle will instead capture a string for
+  each module function and use an evaluator function to rehydrate them.
+  The default evaluator is `eval` on `globalThis`, but can be overridden
+  with the `evaluate` curried runtime option, with `curryRuntimeOptions`.
+  This can make the line numbers in stack traces more useful.
+- `curryRuntimeOptions` (`boolean`, default `false`):
+  By default, the completion value of a bundle is the exports namespace of
+  the entry module after all the modules have executed.
+  This is suitable for shims like `ses`, where the completion value is
+  discarded and any runtime options must be communicated through the
+  global environment, like `process.env` or the presence of a `require` function.
+  Currying runtime options changes the completion value to a function that receives
+  runtime options and then returns the exports namespace of the entry module.
```

Comment:
I don't think "currying" is technically correct here, nor does it communicate to me what's really going on (the indirection from "bundle represents a [non-function] module namespace" to "bundle represents a function that returns a module namespace"). I don't have a better name suggestion, but I wonder if the indirection might be better represented by a sibling function distinct from `makeBundle` rather than an option for it (and I really an wondering; the answer might be "no").

Reply:
Per out-of-band conversation, we found that we can avoid the option name by splitting `makeBundle` into `makeScript` and `makeFunctor`, where the latter provides an opportunity to apply runtime options.